patches/minecraft/net/minecraft/item/ItemStack.java.patch

```diff
@@ -309,4 +309,42 @@
 +    {
 +        return this.field_151002_e;
 +    }
++
++    /**
++     * Modeled after ItemStack.areItemStacksEqual
++     * Uses Item.getNBTShareTag for comparison instead of NBT and capabilities.
++     * Only used for comparing itemStacks that were transferred from server to client using Item.getNBTShareTag.
++     */
++    public static boolean areItemStacksEqualUsingNBTShareTag(ItemStack stackA, ItemStack stackB)
++    {
++        if (stackA.func_190926_b())
++            return stackB.func_190926_b();
++        else
++            return !stackB.func_190926_b() && stackA.isItemStackEqualUsingNBTShareTag(stackB);
++    }
++
++    /**
++     * Modeled after ItemStack.isItemStackEqual
++     * Uses Item.getNBTShareTag for comparison instead of NBT and capabilities.
++     * Only used for comparing itemStacks that were transferred from server to client using Item.getNBTShareTag.
++     */
++    private boolean isItemStackEqualUsingNBTShareTag(ItemStack other)
```

Comment:
I broke up the methods similar to the way vanilla does so that the logic is easier to verify.